### PR TITLE
Small Code Refactor; Saving/Restoring/Evaluating

### PR DESCRIPTION
**Pull request by rbrthogan**
* split out network/inference definition into separate file
* reuse for train/val cycle as well as final model usage e.g. feature extraction / evaluation
* added saving / restoring of model 

* some code duplication introduced; to be addressed in future PRs

Note: I noticed there was changes to the master branch after I made this PR. A lot of these seemed like scrap code for adhoc experiments. I've commented it out but left it there. This should be cleaned up and useful bits kept in a reusable format

### CODE REVIEW

File: network.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    conv2 = conv_layer(input=conv1, num_input_channels=num_filters1,
+                       filter_size=filter_size2, num_filters=num_filters2, name="conv2")
+    conv3 = conv_layer(input=conv2, num_input_channels=num_filters2,
+                       filter_size=filter_size3, num_filters=num_filters3, name="conv3")
+
+    layer_flat, num_features = flatten_layer(conv3)
+
+    fc1 = fc_layer(input=layer_flat, num_inputs=num_features, num_outputs=fc_size, name="fc1")
+
+    logits = fc_layer(input=fc1, num_inputs=fc_size, num_outputs=num_classes, name="fc2")
```
Comment (sree-varma) [inline network.py]: activated_fc1 = tf.nn.relu(fc1). We need to activate the fc1 with ReLU right??

File: jet_images_ml.py [hunk 2]
```diff
@@ -238,33 +164,41 @@ def optimize(num_iterations, starting_iteration=0):
...
+                validate_acc, val_loss = session.run([accuracy, cost], feed_dict=feed_dict_validate)
+                av_validate_acc = av_validate_acc + validate_acc
+                av_val_loss = av_val_loss + val_loss
+            av_val_loss /= valdiation_batches
+            av_validate_acc /= valdiation_batches
             summary_value_val = session.run(merged_summary, feed_dict_validate)
             eval_writer.add_summary(summary_value_val, i)
-            
+
+        if i % saving_freq:
```
Comment (sree-varma) [inline jet_images_ml.py]: Is it if i% saving_freq==0: ??

Comment (rbrthogan) [reply jet_images_ml.py]: correct sorry my mistake

File: network.py (hunk 1 above)
Comment (rbrthogan) [reply network.py]: ah, my mistake. I thought the ReLU was part of the fc_layer function. I will fix this and update now

File: network.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+
+def conv_layer(input, num_input_channels, filter_size, num_filters, name="conv"):
+    with tf.name_scope(name):
+        shape = [filter_size, filter_size, num_input_channels, num_filters]
+        w = tf.Variable(tf.truncated_normal(shape, stddev=0.1), name="W")
+        b = tf.Variable(tf.constant(0.1, shape=[num_filters]), name="B")
+        conv = tf.nn.conv2d(input, w, strides=[1, 1, 1, 1], padding="SAME")
+        act = tf.nn.relu(conv + b)
+        tf.summary.histogram("weights", w)
```
Comment (sree-varma) [inline network.py]: Why are we not adding dropout?

**Comment by rbrthogan** [comment]
No reason, did you have it there before? Didn't mean to remove it.

You can of course add it here and vary the dropout fraction as needed to
improve generalization

On Feb 5, 2018 6:16 PM, "Sreedevi Varma" <notifications@github.com> wrote:

*@sree-varma* commented on this pull request.
------------------------------

In network.py
<https://github.com/sree-varma/Jet-Images/pull/5#discussion_r166060550>:

> @@ -0,0 +1,95 @@
+import tensorflow as tf
+
+"""
+Helper Functions
+"""
+
+
+def conv_layer(input, num_input_channels, filter_size, num_filters,
name="conv"):
+    with tf.name_scope(name):
+        shape = [filter_size, filter_size, num_input_channels, num_filters]
+        w = tf.Variable(tf.truncated_normal(shape, stddev=0.1), name="W")
+        b = tf.Variable(tf.constant(0.1, shape=[num_filters]), name="B")
+        conv = tf.nn.conv2d(input, w, strides=[1, 1, 1, 1], padding="SAME")
+        act = tf.nn.relu(conv + b)
+        tf.summary.histogram("weights", w)

Why are we not adding dropout?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/sree-varma/Jet-Images/pull/5#pullrequestreview-94089047>,
or mute the thread
